docs/guides/usage.md

```diff
@@ -0,0 +1,83 @@
+# How to use the lakeFS file system
+
+This guide contains instructions and code snippets on how to use the lakeFS file system.
+
+## Uploading and downloading files
+
+The arguably most important feature of the file system are file transfers.
+
+### File uploads
+
+To upload a file, you can use the `fs.put()` and `fs.put_file` methods:
+
+```python
+from lakefs_spec import LakeFSFileSystem
+
+fs = LakeFSFileSystem()
+
+# remote path, then local target path.
+fs.put_file("file.txt", "my-repo/my-ref/file.txt")
+```
+
+If you want to upload a whole directory to lakeFS, use the `fs.put` API together with the `recursive=True` switch:
+
+```python
+# structure:
+#   dir/a.txt
+#      /b.yaml
+#      /c.csv
+#      /...
+
+from lakefs_spec import LakeFSFileSystem
+
+fs = LakeFSFileSystem()
+
+# creates the target directory `dir` in the current directory.
+fs.put("dir", "my-repo/my-ref/dir", recursive=True)
+```
+
+!!! info
+
+    This method of file uploading results in two transfers: Once from the client to the lakeFS server, and once from the lakeFS server to the object storage.
+    This can impact performance if very large files are uploaded frequently. To avoid this performance hit, you can also decide to write the file directly to the underlying object storage:
```

Comment:
Only once can be enough, if the files are large enough :sweat_smile:

```suggestion
    This can impact performance if very large files are uploaded. To avoid this performance hit, you can also decide to write the file directly to the underlying object storage:
```

Reply:
Fixed in [8ca0a64](https://github.com/aai-institute/lakefs-spec/pull/156/commits/8ca0a64d1045e89f9db40f7f03c6f693f2f3891a).